### PR TITLE
Lifecycle implementation

### PR DESCRIPTION
**Pull request by namArora3112**
## Description
https://jira.corp.adobe.com/browse/MOB-22349

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: apps/AEPSampleAppNewArchEnabled/android/app/src/main/java/com/AEPSampleAppNewArchEnabled/MainActivity.kt [hunk 1]
```diff
@@ -54,17 +54,18 @@ class MainActivity : ReactActivity() {
...
-    // to do - add lifecycle methods
-//     override fun onResume() {
-//        super.onResume()
-//        MobileCore.setApplication(application);
-//        MobileCore.lifecycleStart(null);
-//    }
+    // Called when the main activity start interacting
+    override fun onResume() {
+        super.onResume()
+        (application as MainApplication).onResume()
```
Comment (yangyansong-adbe) [inline apps/AEPSampleAppNewArchEnabled/android/app/src/main/java/com/AEPSampleAppNewArchEnabled/MainActivity.kt:60]: https://developer.adobe.com/client-sdks/home/base/mobile-core/lifecycle/android/#implementing-global-lifecycle-callbacks
Can we follow the lifecycle implementation outlined in the above public documentation?

File: apps/AEPSampleAppNewArchEnabled/android/app/src/main/java/com/AEPSampleAppNewArchEnabled/MainActivity.kt [hunk 2]
```diff
@@ -54,17 +54,18 @@ class MainActivity : ReactActivity() {
...
+    }
+
+    // Called when the activity pauses
+    override fun onPause() {
+        super.onPause()
+        (application as MainApplication).onPause()
```
Comment (yangyansong-adbe) [inline apps/AEPSampleAppNewArchEnabled/android/app/src/main/java/com/AEPSampleAppNewArchEnabled/MainActivity.kt:66]: Dito

Comment (namArora3112) [reply apps/AEPSampleAppNewArchEnabled/android/app/src/main/java/com/AEPSampleAppNewArchEnabled/MainActivity.kt:66]: @cacheung  can you please discuss with Yangyansong and see what we can do as i was just using your code .

File: apps/AEPSampleAppNewArchEnabled/android/app/src/main/java/com/AEPSampleAppNewArchEnabled/MainActivity.kt (hunk 1 above)
Comment (namArora3112) [reply apps/AEPSampleAppNewArchEnabled/android/app/src/main/java/com/AEPSampleAppNewArchEnabled/MainActivity.kt]: have updated the code accordingly @yangyansong-adbe

File: apps/AEPSampleAppNewArchEnabled/android/app/src/main/java/com/AEPSampleAppNewArchEnabled/MainActivity.kt (hunk 2 above)
Comment (namArora3112) [reply apps/AEPSampleAppNewArchEnabled/android/app/src/main/java/com/AEPSampleAppNewArchEnabled/MainActivity.kt]: have updated the code accordingly @yangyansong-adbe

**Review by cacheung — APPROVED** [review]

**Pull request merged**